StardewArchipelago/GameModifications/CodeInjections/Modded/JunimoShopInjections.cs

```diff
@@ -0,0 +1,224 @@
+        private static JunimoShopGenerator _junimoShopGenerator;
+        private static readonly string Question = "Me love purple thing-a-ma-bobs!  Could give VERY special gift!  You want?";
+        private static readonly string Money = "Weird yellow rocks for {0} {1}.";
+        private static readonly string Dewdrop = "Blue tasties under pretty tree for {0} {1}.";
+        private static readonly string Friendship = "Kiss many people on forehead for {0} {1}.";
+        private static readonly Response Nothing = new("No", "Nothing for now!");
+        private static Dictionary<string, PurpleJunimo> PurpleJunimoOptions {get; set;}
+        private static NPC DaJunimo = new();
+        private static readonly string _junimoDialogueKey = "PurpleJunimoVendor";
+
+        
+        private static readonly List<string> _junimoFirstItems = new(){
+            "Legend", "Prismatic Shard", "Ancient Seeds", "Dinosaur Egg", "Tiny Crop (stage 1)", "Super Starfruit", "Magic Bait"
+        };
+
+        private static readonly Dictionary<string, string> _junimoPhrase = new(){
+            {"Orange", "Look! Me gib tasty \nfor orange thing!" },
+            {"Red", "Give old things \nfor red gubbins!"},
+            {"Grey", "I trade rocks for \n grey what's-its!"},
+            {"Yellow", "I hab seeds, gib \nyellow gubbins!"},
+            {"Blue", "I hab fish! You \ngive blue pretty?"}
+
+        };
+        private static readonly Dictionary<string, string> _firstItemToColor= new(){
+            {"Legend", "Blue"}, {"Prismatic Shard", "Grey"}, {"Dinosaur Egg", "Red"}, {"Ancient Seeds", "Yellow"},
+            {"Tiny Crop (stage 1)", "Orange"}, {"Super Starfruit", "Purple"}, {"Magic Bait", "Purple"}
+        };
+
+        public static void Initialize(IMonitor monitor, IModHelper modHelper, ArchipelagoClient archipelago, ShopStockGenerator shopStockGenerator, StardewItemManager stardewItemManager, JunimoShopGenerator junimoShopGenerator)
+        {
+            _monitor = monitor;
+            _modHelper = modHelper;
+            _archipelago = archipelago;
+            _shopStockGenerator = shopStockGenerator;
+            _stardewItemManager = stardewItemManager;
+            _junimoShopGenerator = junimoShopGenerator;
+        }
+
+
+
+        private static ShopMenu _lastShopMenuUpdated = null;
+        
+        // public override void update(GameTime time)
+        public static bool Update_JunimoWoodsAPShop_Prefix(ShopMenu __instance, GameTime time)
+        {
+            try
+            {
+                // We only run this once for each menu
+                if (__instance.storeContext != "Custom_JunimoWoods")
+                {
+                    return true;
+                }
+                var firstItemForSale = "";
+                if (__instance.forSale.Count == 0) // in case the shop is emptied on a second pass
+                {
+                    firstItemForSale = "";
```

Comment:
This if just does exactly what would happen if it didn't exist, as this is the default value. Just invert the if and don't have an else